api/src/middleware/authenticate.ts

```diff
@@ -45,7 +48,18 @@ export const handler = async (req: Request, _res: Response, next: NextFunction)
 		return next();
 	}
 
-	req.accountability = await getAccountabilityForToken(req.token, defaultAccountability);
+	try {
+		req.accountability = await getAccountabilityForToken(req.token, defaultAccountability);
+	} catch (error) {
+		// Clear the session cookie if the provided token is invalid,
+		// allowing the client to login again
+		if (isDirectusError(error, ErrorCode.InvalidToken) && req.tokenSource === 'cookie') {
+			const env = useEnv();
+			res.clearCookie(env['SESSION_COOKIE_NAME'] as string, SESSION_COOKIE_OPTIONS);
+		}
+
```

Comment:
Wasn't a fan of having it in the request either 👍 But I'd like to be certain about the token source to prevent any (malicious or unintended) logouts via invalid `access_token` queries.

Therefore, I've now combined the `extractToken` and `getAccountability` functions in the `authenticate` middleware, so we can easily access the `source` where we need it.
IMO that structure makes more sense anyway - we do not need to distribute these two functions, which both serve the purpose of authentication, across two middleware.
This also makes testing easier/cleaner.

On that note, we could now also drop `req.token`, as it is not used anywhere in the API. However, there could be extensions that make use of this... Would certainly be a separate PR.

Reply:
> This solution feels like the simplest one

I don't feel this review statement still holds after the latest refactor. While im not against the merging of these 2 middle wares it looks like some other unrelated (maybe unintended) things were refactored here as well. 

Will have to review from scratch again.